pkg/cluster/internal/providers/podman/OWNERS

```diff
@@ -1,13 +1,13 @@
 # See the OWNERS docs at https://go.k8s.io/owners
 
 reviewers:
-  - amwat
   - aojea
   - BenTheElder
 approvers:
-  - amwat
```

Comment:
We already did this elsewhere but missed this file, which caused the bot to assign him to this PR